### PR TITLE
Fix signed requests with escaped urls

### PR DESCRIPTION
**Pull request by achamayou**
Fix for #2592. The signature verification code previously re-assembled decoded path and query string to construct the canonical payload. This change makes it use the original URL instead.

- [x] Confirm we need to drop fragment and drop it: we don't and we won't, details on ticket.
- [x] CHANGELOG

### CODE REVIEW

File: tests/e2e_logging.py [hunk 1]
```diff
@@ -746,6 +746,13 @@ def test_forwarding_frontends(network, args):
         ack = network.consortium.get_any_active_member().ack(backup)
         check_commit(ack)
 
+    samples = [
+        "this=that",
+        "this=that&other=with spaces",
+        "this with spaces=with spaces",
+        'arg=This has many@many many \\% " AWKWARD :;-=?!& characters %20%20',
+    ]
+
```
Comment (eddyashton) [inline tests/e2e_logging.py:755]:
```suggestion
```
Believe this is now defined on 768.

File: tests/e2e_logging.py [hunk 2]
```diff
@@ -793,6 +799,53 @@ def test_forwarding_frontends(network, args):
...
+@reqs.description("Testing signed queries with escaped queries")
+@reqs.at_least_n_nodes(2)
+def test_signed_escapes(network, args):
+    samples = [
+        "this=that",
+        "this=that&other=with spaces",
+        "this with spaces=with spaces",
+        'arg=This has many@many many \\% " AWKWARD :;-=?!& characters %20%20',
+    ]
+
```
Comment (eddyashton) [inline tests/e2e_logging.py:811]:
```suggestion
```
Now defined on 815.

File: tests/e2e_logging.py [hunk 3]
```diff
@@ -793,6 +799,53 @@ def test_forwarding_frontends(network, args):
     return network
 
 
+@reqs.description("Testing signed queries with escaped queries")
+@reqs.at_least_n_nodes(2)
+def test_signed_escapes(network, args):
```
Comment (eddyashton) [inline tests/e2e_logging.py:804]: This duplicates a lot of `test_forwarding_frontends`, which could be factored out into something like `def escaped_query_tests(client, url_base)`.

**Review by jumaffre — APPROVED** [review]

**Review by eddyashton — APPROVED** [review]

File: tests/e2e_logging.py (hunk 3 above)
Comment (achamayou) [reply tests/e2e_logging.py:799]: Yes, good point.

**Pull request merged**